integrationtests/all-embedded-query-it/src/test/java/org/infinispan/all/embeddedquery/QueryDslConditionsTest.java

```diff
@@ -1724,6 +1786,55 @@ public void testUnorderedPagination8() throws Exception {
       assertEquals(2, list.size());
    }
 
+   @Test
+   public void testGroupBy() throws Exception {
+      QueryFactory qf = getQueryFactory();
+
+      Query q = qf.from(getModelFactory().getUserImplClass())
+            .select("name")
+            .groupBy("name")
+            .orderBy("name", SortOrder.ASC)
+            .build();
+
+      List<Object[]> list = q.list();
+      assertEquals(2, list.size());
+      assertEquals(1, list.get(0).length);
+      assertEquals("John", list.get(0)[0]);
+      assertEquals(1, list.get(1).length);
+      assertEquals("Spider", list.get(1)[0]);
+   }
+
+   @Test
+   public void testGroupBy2() throws Exception {
+      QueryFactory qf = getQueryFactory();
+
+      Query q = qf.from(getModelFactory().getUserImplClass())
+            .select(Expression.sum("age"))
+            .groupBy("name")
+            .orderBy("name", SortOrder.ASC)
+            .build();
+
+      List<Object[]> list = q.list();
+      assertEquals(2, list.size());
+      assertEquals(1, list.get(0).length);
+      assertEquals(22, list.get(0)[0]);
+      assertEquals(1, list.get(1).length);
+      assertEquals(null, list.get(1)[0]);
+   }
+
+   @Test(expected = ParsingException.class)
+   public void testGroupBy3() throws Exception {
+      QueryFactory qf = getQueryFactory();
+
+      Query q = qf.from(getModelFactory().getUserImplClass())
+            .select("name")
+            .groupBy("name")
+            .orderBy("surname", SortOrder.ASC)
+            .build();
+
```

Comment:
What about doing aggregation in a filtered query? Would something conceptually similar to ```SELECT name FROM person WHERE name.startsWith("a") GROUP BY name``` work? It seems there is no test for it